server/models/Expense.ts

```diff
@@ -932,6 +933,13 @@ Expense.init(
       defaultValue: false,
       allowNull: false,
     },
+
+    reference: {
+      type: DataTypes.STRING,
+      set(reference: string | null) {
+        this.setDataValue('reference', isNil(reference) ? null : reference.trim());
```

Comment:
Unless we want to support empty strings, I would consolidate:

```suggestion
        this.setDataValue('reference', reference?.trim() || null);
```